src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -105,7 +100,24 @@ public Endpoint(String id, Conference conference)
         super(conference, id);
 
         this.logger = Logger.getLogger(classLogger, conference.getLogger());
-        this.messageTransport = new EndpointMessageTransport(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected EndpointMessageTransport createTransportChannel()
+    {
+        return new EndpointMessageTransport(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EndpointMessageTransport getMessageTransport()
+    {
+        return (EndpointMessageTransport) super.getMessageTransport();
```

Comment:
what's the need for the cast here?

Reply:
Because super returns the abstract class

Reply:
but that type will declare all the methods, right?  the instance will be the subclass, but shouldn't having it typed as the abstract class be fine?

Reply:
Endpoint.java uses methods specific to the subclass.

Reply:
what are the methods in `EndpointMessageTransport` that `Endpoint` uses which `AbstractEndpointMessageTransport` doesn't define?

Reply:
`getSctpConnection`, `setSctpConnection`, `onWebSocketClose`, `onWebSocketConnect`, `onWebSocketText`,

Reply:
those all seem like generic methods though, why not define those in the top class/interface?

Reply:
They are not generic, SCTP and web sockets are only used by the Endpoint class (and not by the other AbstractEndpoint implementation)